### PR TITLE
[Model] PCT

### PR DESCRIPTION
**Pull request by sangyx**
## Description

This is a reproduction of the paper: [PCT: Point cloud transformer](http://arxiv.org/abs/2012.09688).

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes

- [x] Implement PCT for classification and part segmentation.
- [x] Fix the error of the opt in point transformer

### CODE REVIEW

File: examples/pytorch/pointcloud/pct/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+PCT
+====
+
+This is a reproduction of the paper: [PCT: Point cloud transformer](http://arxiv.org/abs/2012.09688).
+
+# Performance
```
Comment (hetong007) [inline examples/pytorch/pointcloud/pct/README.md:6]: Please specify the difference from the paper description, and also include the results from an exact reproduction.

File: examples/pytorch/pointcloud/pct/helper.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            h = conv(h)
+            h = bn(h)
+            h = F.relu(h)
+        h = torch.max(h, 2)[0]
+        feat_dim = h.shape[1]
+        h = h.permute(0, 2, 1).reshape(-1, feat_dim)
+        return {'new_feat': h}
+
+
+class GraphOp(nn.Module):
```
Comment (hetong007) [inline examples/pytorch/pointcloud/pct/helper.py:142]: Giving it the name `TransitionDown` is better than this information-less `GraphOp`.

**Review by hetong007 — APPROVED** [review]
Some minor comments. In general LGTM.

File: examples/pytorch/pointcloud/pct/helper.py (hunk 2 above)
Comment (sangyx) [reply examples/pytorch/pointcloud/pct/helper.py:142]: updated

File: examples/pytorch/pointcloud/pct/README.md (hunk 1 above)
Comment (sangyx) [reply examples/pytorch/pointcloud/pct/README.md:6]: updated

**Pull request merged**